### ISSUE TITLE
Question about Numer.ai notebook

### ISSUE THREAD

**Issue by jvmancuso** [QuantScientist/Deep-Learning-Boot-Camp#1]
For reference, I'm looking at [this notebook](https://github.com/QuantScientist/Deep-Learning-Boot-Camp/blob/master/day%2002%20PyTORCH%20and%20PyCUDA/PyTorch/18-PyTorch-NUMER.AI-Binary-Classification-BCELoss.ipynb).

I am really confused by the architecture used in the Net2 class.  Can you explain a little bit about how you arrived at it?  It almost seems like the result of an automated architecture search, however the performance seems too poor for that to be true.  It seems to me that the model is severely underfitting, and I can see several reasons for this.  I'd be willing to reconstruct a better example if you're accepting PRs.

**Comment by QuantScientist** [comment]
Hi @jvmancuso !
Thank you very much for your comment,  I updated the notebook to reflect the fact that the NN architecture is not optimized. This was an attempt to show an end to end working example on a real use case. 

The architecture includes many commonly used layers, (once again) for depicting their use, and they do not necessarily form a   "logical" NN. I am working on a better NN architecture and in addition, also on a CNN based PyTorch example using the same data set : https://github.com/QuantScientist/Deep-Learning-Boot-Camp/blob/master/day%2002%20PyTORCH%20and%20PyCUDA/PyTorch/31-PyTorch-using-CONV1D-on-one-dimensional-data.ipynb

Please go a head and construct a better example, I will, of course, accept a PR's. 

Thanks,

**Comment by QuantScientist** [comment]
I added you as a collaborator.

**Comment by jvmancuso** [comment]
@QuantScientist Question on the data: what do these instances/features represent?  Is there some sort of sequential structure to them?  Would appreciate more details on the data.

**Comment by sovaa** [comment]
Hi,

Thanks for sharing the notebook! I also participate in the competition but I haven't had much experience with NNs before except some PoC's in keras, so was both valuable in terms of learning a bit more about NN architectures for BC and how to do it with pytorch. 

Like @jvmancuso mentioned, it would be great to see some comments in the notebook on the reasons for some of the architectural choices; e.g. why in `Net2()` you chose `LeakyRELU` for all hidden layers (e.g. what's the benefit compared to `tanh` or `relu`), and in the next NN example after `Net2()` the activations are `tanh`, `tanh`, `relu`, `tanh`, `relu`. Also the log loss printed in the notebook, `0.6939`, is worse than random, `0.6931` (I know you said it's not optimized and just showing the general idea of how to do it, just saying!).

Looking forward to more updates!

**Comment by QuantScientist** [comment]
@sovaa 
My pleasure, 
The choices were mostly random and the result of experimentation, and as you indicated the actual scores are not good. I gained a much better score with classical ML (I linked a notebook for that in the tutorial). 

I shall add more comments to explain, and will update once I have a better working NN architecture. 
Meanwhile, you can also take a look at the CNN version here: https://github.com/QuantScientist/Deep-Learning-Boot-Camp/blob/master/day%2002%20PyTORCH%20and%20PyCUDA/PyTorch/55-PyTorch-using-CONV1D-on-one-dimensional-data-CNN.ipynb which is also work in progress and not optimized but it can get you started. 

Do let me know if you need anything else and feel free to post PR's 

Best,

**Comment by QuantScientist** [comment]
@sovaa  and @jvmancuso and @nathanie
I updated the Jupyter notebook, it now includes the actual score on the NumerAI board and many comments which were added. It is far from being perfect, and there is much room for improvement as noted above.  I hope you can improve it with me. 
https://github.com/QuantScientist/Deep-Learning-Boot-Camp/blob/master/day%2002%20PyTORCH%20and%20PyCUDA/PyTorch/18-PyTorch-NUMER.AI-Binary-Classification-BCELoss.ipynb

Best, 
Shlomo.
